restrict_useradd.sh

```diff
@@ -31,6 +31,6 @@ if [ $uid -eq 0 ]; then
 elif [ $gid -eq 0 ]; then
     echo "Refusing to use a gid of 0"
     exit 1
-else
+elif [ ! $(getent passwd pokyuser) ]; then
     useradd -N -g $gid -m $skelarg -o -u $uid "$username"
```

Comment:
If we don't do anything at all if the user already exists, the user won't get added to the group. This usually wouldn't matter, but would matter in the case where the files in the workdir are accessible based on the group and not the user.

Also interesting but not as important, if the caller specified a skeleton directory it won't get used. But currently I nothing is using the skelarg. It was going to be used by https://github.com/crops/poky-container/pull/2, which is sort of dead unless someone asks for it.

Reply:
Thanks for the more detailed explanation; I think I understand where you are coming from now.

I'm not sure how to handle the skel argument since if the user was already created then the skel was already used to make the home path, etc.  Changing it after the fact wouldn't do anything (I think...).

As for the group membership, an `else` to `usermod -aG $gid $uid` would work and will not throw an error code if the user is already in that group.  Does that sound good?

Reply:
I think that might be fine for now.

But we could possibly just do a userdel and always call useradd. But that might affect things if you create persistent containers.

On that note, I may create an issue for adding tests for all of the user stuff when a container runs a second time. Because it is probably a reasonable assumption that a user shouldn't actually be created the second time the container is run.